Update _index.md (#804)
mattermost/mattermost-developer-documentation: site/content/integrate/apps/_index.md

Read the [quick start guide]({{< ref  "quick-start-go" >}}) to learn how to write your first app.

## FAQ

### When would you build an app vs a custom slash command vs a webhook vs a plugin?

> That depends on your use case, as they each have benefits. Apps can be advertised, shared, and installed by others via the Marketplace and users can get started with a slash command. Comparatively, a custom slash and webhooks command requires a lot of configuration, is very rigid - and is limited to your own server.

Reviewer: > Apps can be advertised, shared, and installed by others via the Marketplace 

Not really right now. It's unclear how the Marketplace will allow installation for Apps besides the Mattermost build ones.

Reviewer: Please also see https://github.com/mattermost/mattermost-developer-documentation/pull/804#discussion_r622019565 on the selling point of apps.

> That depends on your use case, as they each have benefits. Apps can be advertised, shared, and installed by others via the Marketplace and users can get started with a slash command. Comparatively, a custom slash and webhooks command requires a lot of configuration, is very rigid - and is limited to your own server.

Reviewer: What does "and is limited to your own server" mean?

> That depends on your use case, as they each have benefits. Apps can be advertised, shared, and installed by others via the Marketplace and users can get started with a slash command. Comparatively, a custom slash and webhooks command requires a lot of configuration, is very rigid - and is limited to your own server.

Author: ```suggestion
That depends on your use case, as they each have benefits. Apps can be advertised, shared, and installed by others via the Marketplace and users can get started with a slash command. Comparatively, a custom slash and webhooks command requires a lot of configuration and is less interactive.
```

## FAQ

### When would you build an app vs a custom slash command vs a webhook vs a plugin?

> That depends on your use case, as they each have benefits. Apps can be advertised, shared, and installed by others via the Marketplace and users can get started with a slash command. Comparatively, a custom slash and webhooks command requires a lot of configuration and is less interactive.

Reviewer 2: ```suggestion
That depends on your use case, as they each have benefits. Apps can be advertised, shared, and installed by others via the Marketplace. Comparatively, a custom slash command requires some configuration and is less interactive.

The built-in incoming webhook is rigid and requires specific fields to be posted to it's URL, but some external systems don't allow the customization of the fields included in the webhook payload.   
Compared to built-in webhooks, An App can be customized to receive an HTTP webhook posting from another system, and can process the incoming data then augment it or make an actionable message in a channel.  

A plugin should be used when you need to alter the React UX in Mattermost or have an application that requires low-latency with the server (such as replacing characters in any message). Currently, plugins have several UX hooks that Apps cannot access, however we plan to add/migrate more UX hooks into the Apps Framework over time. 
```

### What's the difference between the apps framework and the plugin framework?

The App framework provides a few differences from plugins.

- We’ve made our interactive elements easier to use and develop.
- Works on both desktop and mobile clients without any extra code.
- You host the app code or build for Matterhost and distribute through the Marketplace.

### What language should I use to write apps?

> Any language you want. However, the only language with an official driver right now is Go.

Reviewer 2: ```suggestion
Any language you want. However, the only language with an official driver right now is Go.  we are planning to release other drivers in the future in conjunction with our community.
```

## FAQ

### When would you build an app vs a custom slash command vs a webhook vs a plugin?

> That depends on your use case, as they each have benefits. Apps can be advertised, shared, and installed by others via the Marketplace. Comparatively, a custom slash command requires some configuration and is less interactive.

Reviewer: > Apps can be advertised, shared, and installed by others via the Marketplace.

That is not the case right now.

> The built-in incoming webhook is rigid and requires specific fields to be posted to it's URL, but some external systems don't allow the customization of the fields included in the webhook payload. Compared to built-in webhooks, An App can be customized to receive an HTTP webhook posting from another system, and can process the incoming data then augment it or make an actionable message in a channel.

Reviewer: 2/5 that customizable webhooks are not an outstanding feature for apps. You can do the same with a simple proxy.

> A plugin should be used when you need to alter the React UX in Mattermost or have an application that requires low-latency with the server (such as replacing characters in any message). Currently, plugins have several UX hooks that Apps cannot access, however we plan to add/migrate more UX hooks into the Apps Framework over time.

Reviewer: What does "Reacht UX" mean?

### What's the difference between the apps framework and the plugin framework?

The App framework provides a few differences from plugins.

- We’ve made our interactive elements easier to use and develop.
- Works on both desktop and mobile clients without any extra code.

> - You host the app code or build for Matterhost and distribute through the Marketplace.

Reviewer 3: Is Matterhost a real thing?

### What language should I use to write apps?

> Any language you want. However, the only language with an official driver right now is Go.  we are planning to release other drivers in the future in conjunction with our community.

Reviewer 3: ```suggestion
Any language you want. We currently have an official driver for Go. We are planning to release other drivers in the future in conjunction with our community.
```
